src/dc_tools.rs

```diff
@@ -199,10 +204,15 @@ pub(crate) fn dc_create_outgoing_rfc724_mid(grpid: Option<&str>, from_addr: &str
         .find('@')
         .map(|k| &from_addr[k..])
         .unwrap_or("@nohost");
-    match grpid {
-        Some(grpid) => format!("Gr.{}.{}{}", grpid, dc_create_id(), hostname),
-        None => format!("Mr.{}.{}{}", dc_create_id(), dc_create_id(), hostname),
-    }
+    let id_tmp: String;
```

Comment:
The motivation for the change is not rust specific, idea is to use only one "format!" line, because format is the same and differs in parameters only.

Reply:
i see.